cibuild.cmd

```diff
@@ -59,6 +54,18 @@ REM Ensure the binaries directory exists because msbuild can fail when part of t
 set bindir=%RoslynRoot%Binaries
 if not exist "%bindir%" mkdir "%bindir%" || goto :BuildFailed
 
+if defined testBuildCorrectness (
+    powershell -noprofile -executionPolicy RemoteSigned -file "%RoslynRoot%\build\scripts\test-build-correctness.ps1" %RoslynRoot% "%bindir%\%BuildConfiguration%" || goto :BuildFailed
+    call :TerminateBuildProcesses
+    exit /b 0
+)
+
+powershell -noprofile -executionPolicy RemoteSigned -file "%RoslynRoot%\build\scripts\check-branch.ps1" || goto :BuildFailed
```

Comment:
Since this is such a quick check, it's kind of nice to have it be before the restore.  Also, it would be weird for every other leg to fail with a `check-branch` violation, but NOT the build correctness one.